### PR TITLE
[FRONTEND][TENSORFLOW] Support Unstack and Split

### PR DESCRIPTION
**Pull request by alexeyr**

### CODE REVIEW

**Comment by alexeyr** [comment]
Test failure should be spurious: the same test failed for other pull requests submitted at similar time, including a doc fix.

File: nnvm/python/nnvm/frontend/tensorflow.py [hunk 1]
```diff
@@ -1137,21 +1191,30 @@ def from_tensorflow(self, graph, layout="NHWC", shape=None):
...
+
                     if i in self._nodes:
-                        inputs.append(self._nodes[i])
-                        input_shapes[self._nodes[i]] = self._output_shapes[i]
+                        tvm_n = self._nodes[i]
+                        outputs = tvm_n.list_output_names()
+                        if len(outputs) > 1:
+                            tvm_n = tvm_n[num_layer]
+                        inputs.append(tvm_n)
+                        input_shapes[tvm_n] = self._output_shapes[i]
```
Comment (Rasterer) [inline nnvm/python/nnvm/frontend/tensorflow.py]: Should it be the following for multiple outputs cases?
`input_shapes[tvm_n] = (self._output_shapes[i])[num_layer]`

**Comment by alexeyr** [comment]
I've run into a curious issue with `unstack`. It boils down to this difference: 

In Tensorflow, 

    In [30]: tf.squeeze(tf.placeholder("int32", [1]),0).shape
    Out[30]: TensorShape([])

But in TVM,

    In [31]: topi.squeeze(tvm.placeholder((1,)),0).shape
    Out[31]: [1]

The expected result of `squeeze` is to drop by one dimension, but in this case it doesn't.
So when I tried to test `tf.stack(tf.unstack(in_data, axis=0), axis=0)` where `in_data` has shape `(6,0)`, Tensorflow gives the expected result but TVM gives result of shape `(6,1)`.

So, does NNVM support representing scalars with shape `()` instead of `(1,)`? I tried to use `split_item[0]` in https://github.com/dmlc/tvm/pull/2105/commits/d2759720a6a354acbc9c2cbc9323e016bf2620bb#diff-3845514369cc735d96dd63a274a21471R815, but that doesn't work.

Two alternatives: 
1. We could error when trying to `unstack` or `squeeze` 1d tensors. 
2. Mark them with an attribute indicating they "should" be `()` and handle it in `_pack` and `_expand_dims` (do any other operators need it?)

File: nnvm/python/nnvm/frontend/tensorflow.py (hunk 1 above)
Comment (alexeyr) [reply nnvm/python/nnvm/frontend/tensorflow.py]: Yes, I think so. I've added a commit which fixes this, but discovered a problem while trying to create a test case distinguishing them.

**Comment by alexeyr** [comment]
The problem should be fixed now. @srkreddy1238 @siju-samuel seem to be most familiar with this frontend, please review.

**Comment by alexeyr** [comment]
Rebased on master.

**Comment by srkreddy1238** [comment]
@alexeyr  I see two open PR's for Split (other #2123 ).
To give an equal opportunity can you add SplitV and unstack on top of #2123 ?

**Comment by alexeyr** [comment]
By rebasing on top on it? OK, will do tomorrow.

**Comment by alexeyr** [comment]
@srkreddy1238 Done.

**Comment by srkreddy1238** [comment]
@alexeyr pls rebase against latest master.

**Comment by alexeyr** [comment]
@srkreddy1238 Rebased.

File: nnvm/python/nnvm/frontend/tensorflow.py [hunk 2]
```diff
@@ -1162,24 +1218,32 @@ def from_tensorflow(self, graph, layout="NHWC", shape=None, outputs=None):
...
                             tensor_slot = int(tensor_name[1]) if len(tensor_name) > 1 else 0
                             in_sym = in_sym[tensor_slot]
-                            input_shape = (self._output_shapes[node_name])[tensor_slot]
+                            input_shape = self._output_shapes[node_name][tensor_slot]
                         else:
+                            tensor_slot = 0
                             input_shape = self._output_shapes[node_name][0]
                         inputs.append(in_sym)
                         input_shapes[in_sym] = [input_shape]
+                        # This means the node is 1d in NVM and 0d in TF.
```
[the remark this answers is not in the document]
Comment (alexeyr) [reply nnvm/python/nnvm/frontend/tensorflow.py]: Fixed.

**Comment by tqchen** [comment]
Thanks, @srkreddy1238 @alexeyr , this is merged

**Pull request merged**